### PR TITLE
Feature/generating data

### PR DESCRIPTION
**Pull request by kovpaka**

### CODE REVIEW

File: MoneyManager/Repositories/BaseRepository.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        {
+            DbSet.AddRange(entities);
+        }
+
+        public void Update(TEntity entity)
+        {
+            DbSet.Update(entity);
+        }
+
+        protected decimal CalculateBalance(Guid userId)
```
Comment (siarheikarabko) [inline MoneyManager/Repositories/BaseRepository.cs:58]: Please, rename it to `GetUserBalance`

File: MoneyManager/Repositories/BaseRepository.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        {
+            decimal balance = (from transaction in context.Transactions
```
Comment (siarheikarabko) [inline MoneyManager/Repositories/BaseRepository.cs:60]: Consider using here LINQ extensions only and probably you do not need to create an anonym object.

File: MoneyManager/Repositories/Interfaces/ITransactionRepository.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections;
+using MoneyManager.Entities;
+using MoneyManager.Enums;
+
+namespace MoneyManager.Repositories.Interfaces
+{
+    interface ITransactionRepository : IRepository<Transaction>
+    {
+        void DeleteTransactionsInCurrentMount();
```
Comment (siarheikarabko) [inline MoneyManager/Repositories/Interfaces/ITransactionRepository.cs:10]: Should it be **Month**?

File: MoneyManager/Repositories/Interfaces/ITransactionRepository.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        IEnumerable GetOrderedTransactionList(Guid userId);
+
```
Comment (siarheikarabko) [inline MoneyManager/Repositories/Interfaces/ITransactionRepository.cs:12]: Please, be consistent in formatting (compare this interface for example with IRepository.

File: MoneyManager/Repositories/Interfaces/IRepository.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections.Generic;
+
+namespace MoneyManager.Repositories.Interfaces
+{
+    interface IRepository<TEntity> where TEntity : class
+    {
+        void Remove(TEntity entityToDelete);
+        void Remove(object id);
```
Comment (siarheikarabko) [inline MoneyManager/Repositories/Interfaces/IRepository.cs:8]: Why is the argument **object**?

File: MoneyManager/Repositories/TransactionRepository.cs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+                    transaction.Comment
+                }).AsNoTracking().ToList();
+          
+            return result;
+        }
+
+        public IEnumerable GetIncomeAndExpensesForSelectedPeriod(Guid userId, DateTime startDate, DateTime endDate)
+        {
+            List<Transaction> selectedTransactions = (from transaction in DbSet
+                where transaction.Asset.UserId == userId &&
```
Comment (siarheikarabko) [inline MoneyManager/Repositories/TransactionRepository.cs:47]: Please, fix the indents.

File: MoneyManager/Repositories/UserRepository.cs [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+        public User GetByEmail(string email)
+        {
+            return DbSet.AsNoTracking().FirstOrDefault(u => u.Email == email);
+        }
+
+        public IEnumerable GetSortedList()
+        {
+            var result = (
+                            from user in DbSet
```
Comment (siarheikarabko) [inline MoneyManager/Repositories/UserRepository.cs:22]: Something wrong with the formatting.

File: MoneyManager/Repositories/TransactionRepository.cs [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+                .ThenInclude(t => t.ParentCategory)
+                .Where(transaction => transaction.Date.Month == DateTime.Now.Month && 
+                                      transaction.Category.Type == categoryType)
+                .Select(t => t);
+            
+            List<CategoriesTransactionsAmountProjection> categoryWithChildrenAmount = selectedTransactions
+                .Where(transaction => transaction.Category.ParentId != null)
+                .AsEnumerable()
+                .GroupBy(t => new { t.Category.ParentId })
+                .Select(transaction =>
```
Comment (siarheikarabko) [inline MoneyManager/Repositories/TransactionRepository.cs:99]: please, rename it to transactionsGroup

File: MoneyManager/Repositories/TransactionRepository.cs [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+                {
+                    var name = transaction
+                        .Select(t => t.Category.ParentCategory.Name)
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply MoneyManager/Repositories/TransactionRepository.cs:102]: No, I calculate the sum of all children transactions of the parent category and then output it with the parent category name

File: MoneyManager/Repositories/TransactionRepository.cs [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+                    return new CategoriesTransactionsAmountProjection()
+                    {
+                        Name = name,
+                        Amount = transaction.Sum(t => t.Amount)
+                    };
+                }).ToList();
+
+            List<CategoriesTransactionsAmountProjection> singleCategories = await selectedTransactions
+                .Where(transaction => transaction.Category.ParentId == null)
+                .GroupBy(t => t.Category.Name )
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply MoneyManager/Repositories/TransactionRepository.cs:114]: I think so. I don't know how to deal with it differently.

File: MoneyManager/DataGenerators/DataGenerator.cs [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+            this.unitOfWork = unitOfWork;
+        }
+
+        public async Task GenerateDate()
+        {
+            await using IDbContextTransaction dataTransaction = unitOfWork.BeginTransaction();
+            try
+            {
+                await GenerateUsers();
+               await GenerateAssets();
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply MoneyManager/DataGenerators/DataGenerator.cs]: Fixed

File: MoneyManager/Contexts/ApplicationContext.cs [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+            builder.SetBasePath(Directory.GetCurrentDirectory());
+            builder.AddJsonFile("appsettings.json");
+            IConfigurationRoot config = builder.Build();
+            string connectionString = config.GetSection("ConnectionString").Value;
+            optionsBuilder.UseSqlServer(connectionString);
+        }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<User>()
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply MoneyManager/Contexts/ApplicationContext.cs]: Done

File: MoneyManager/Entities/Types/HEXColor.cs [hunk 13]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace MoneyManager.Entities.Types
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply MoneyManager/Entities/Types/HEXColor.cs]: Fixed

File: MoneyManager/Repositories/TransactionRepository.cs [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+                }).Distinct().ToList();
+
+            return result;
+        }
+
+        public async Task<IList<CategoriesTransactionsAmountProjection>> GetCategoriesTransactionsAmount(Guid userId, CategoryType categoryType)
+        {
+            IQueryable<Transaction> selectedTransactions = DbSet
+                .Include(transaction => transaction.Category)
+                .ThenInclude(t => t.ParentCategory)
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply MoneyManager/Repositories/TransactionRepository.cs]: It's useful, thanks, fixed

File: MoneyManager/Repositories/TransactionRepository.cs [hunk 15]
```diff
@@ -0,0 +1,130 @@
...
+                .Where(transaction => transaction.Date.Month == DateTime.Now.Month && 
+                                      transaction.Category.Type == categoryType)
+                .Select(t => t);
+            
+            List<CategoriesTransactionsAmountProjection> categoryWithChildrenAmount = selectedTransactions
```
[the remark this answers is not in the document]
Comment (kovpaka) [reply MoneyManager/Repositories/TransactionRepository.cs]: Fixed

**Pull request merged**